app/experimenter/nimbus-ui/src/components/ChangeApprovalOperations/index.tsx

```diff
@@ -49,30 +57,55 @@ export const ChangeApprovalOperations: React.FC<
   startRemoteSettingsApproval,
   children,
 }) => {
-  let defaultUIState = ChangeApprovalOperationsState.None;
-  if (rejectionEvent) {
-    defaultUIState = ChangeApprovalOperationsState.Rejected;
-  } else if (timeoutEvent) {
-    defaultUIState = canReview
-      ? ChangeApprovalOperationsState.ApproveOrReject
-      : ChangeApprovalOperationsState.ApprovalPending;
-  } else if (approvalEvent) {
-    defaultUIState = canReview
-      ? ChangeApprovalOperationsState.RemoteSettingsPending
-      : ChangeApprovalOperationsState.ApprovalPending;
-  } else if (reviewRequestEvent) {
-    defaultUIState = canReview
-      ? ChangeApprovalOperationsState.ApproveOrReject
-      : ChangeApprovalOperationsState.ApprovalPending;
-  }
+  const uiStateDependencies = [
+    canReview,
+    reviewRequestEvent,
+    approvalEvent,
+    rejectionEvent,
+    timeoutEvent,
+  ];
+
+  const defaultUIState = useMemo(() => {
+    switch (
+      determineLatestChangelogEventKind({
+        reviewRequestEvent,
+        approvalEvent,
+        rejectionEvent,
+        timeoutEvent,
+      })
+    ) {
+      case EventKinds.Rejection:
+        return ChangeApprovalOperationsState.Rejected;
+      case EventKinds.Approval:
+        return canReview
+          ? ChangeApprovalOperationsState.RemoteSettingsPending
+          : ChangeApprovalOperationsState.ApprovalPending;
+      case EventKinds.Request:
+        return canReview
+          ? ChangeApprovalOperationsState.ApproveOrReject
+          : ChangeApprovalOperationsState.ApprovalPending;
+      case EventKinds.Timeout:
+        return canReview
+          ? ChangeApprovalOperationsState.ApproveOrReject
+          : ChangeApprovalOperationsState.ApprovalPending;
+      default:
+        return ChangeApprovalOperationsState.None;
+    }
+  }, uiStateDependencies);
+
   const [uiState, setUIState] = useState<ChangeApprovalOperationsState>(
     defaultUIState,
   );
-  const resetDraftUIState = useCallback(() => setUIState(defaultUIState), [
+
+  const resetUIState = useCallback(() => setUIState(defaultUIState), [
     defaultUIState,
     setUIState,
   ]);
 
+  // Whenever any of the UI default state dependencies change (i.e. via
+  // polling or refetch), override and reset the state.
+  useEffect(() => resetUIState(), uiStateDependencies);
```

Comment:
Ran into a small bug fixed by this, trying to work out how to trigger it in tests to ensure it's exercised.